presto-main/src/main/java/com/facebook/presto/operator/scalar/UnitFunctions.java

```diff
@@ -0,0 +1,168 @@
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, e);
+        }
+    }
+
+    @Description("succinct duration string of milliseconds")
+    @ScalarFunction("succinct_millis")
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice succinctMillis(@SqlType(StandardTypes.BIGINT) long value)
+    {
+        return succinctMillis((double) value);
+    }
+
+    @Description("succinct duration string of milliseconds")
+    @ScalarFunction("succinct_millis")
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice succinctMillis(@SqlType(StandardTypes.DOUBLE) double value)
+    {
+        try {
+            Duration duration = Duration.succinctDuration(value, MILLISECONDS);
+            return utf8Slice(duration.toString());
+        }
+        catch (IllegalArgumentException e) {
+            // When value is negative, etc.
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, e);
+        }
+    }
+
+    @Description("succinct data size string")
+    @ScalarFunction("succinct_data_size")
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice succinctDataSize(@SqlType(StandardTypes.DOUBLE) double value, @SqlType(StandardTypes.VARCHAR) Slice unit)
+    {
+        try {
+            DataSize dataSize = DataSize.succinctDataSize(value, valueOfUnit(unit.toStringUtf8()));
+            return utf8Slice(dataSize.toString());
+        }
+        catch (IllegalArgumentException e) {
+            // When value is negative, NaN, etc.
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, e);
+        }
+    }
+
+    @Description("succinct byte size string")
+    @ScalarFunction("succinct_bytes")
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice succinctBytes(@SqlType(StandardTypes.BIGINT) long value)
+    {
+        try {
+            DataSize dataSize = DataSize.succinctBytes(value);
+            return utf8Slice(dataSize.toString());
+        }
+        catch (IllegalArgumentException e) {
+            // When value is negative, etc.
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, e);
+        }
+    }
+
+    @Description("succinct byte size string")
+    @ScalarFunction("succinct_bytes")
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice succinctBytes(@SqlType(StandardTypes.DOUBLE) double value)
+    {
+        try {
+            DataSize dataSize = DataSize.succinctDataSize(value, BYTE);
+            return utf8Slice(dataSize.toString());
+        }
+        catch (IllegalArgumentException e) {
+            // When value is negative, etc.
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, e);
+        }
+    }
+
+    private static TimeUnit valueOfTimeUnit(String timeUnitString)
+    {
+        try {
+            return Duration.valueOfTimeUnit(timeUnitString);
+        }
+        catch (IllegalArgumentException e) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "TimeUnit string must be one of [ns, us, ms, s, m, h, d]");
```

Comment:
We should pass `e` as the cause to `PrestoException` constructor. Please also fix in other places like below.